go/cmd/dolt/commands/push.go

```diff
@@ -88,6 +89,53 @@ func (cmd PushCmd) Exec(ctx context.Context, commandStr string, args []string, d
 	help, usage := cli.HelpAndUsagePrinters(cli.CommandDocsForCommandString(commandStr, pushDocs, ap))
 	apr := cli.ParseArgsOrDie(ap, args, help)
 
+	if apr.Contains(cli.AllFlag) {
+		return cmd.pushAllBranches(ctx, apr, dEnv, usage)
+	}
+
+	return cmd.doPush(ctx, apr, dEnv, usage)
+}
+
+func (cmd PushCmd) pushAllBranches(ctx context.Context, apr *argparser.ArgParseResults, dEnv *env.DoltEnv, usage cli.UsagePrinter) int {
+	branches, err := dEnv.DoltDB.GetBranches(ctx)
+	if err != nil {
+		verr := errhand.BuildDError("failed to fetch branches").AddCause(err).Build()
+		return HandleVErrAndExitCode(verr, usage)
+	}
+
+	if len(apr.Args) > 1 {
+		verr := errhand.BuildDError("only <remote> may be specified with --all").SetPrintUsage().Build()
+		return HandleVErrAndExitCode(verr, usage)
+	}
+
+	remoteName := "origin"
```

Comment:
Is this what git does? Can't tell from the docs.

Normally `git push` without a remote named will push to the upstream tracking branch, but that can't be the case with `--all`